src/Nethermind/Nethermind.Api/NethermindApi.cs

```diff
@@ -132,17 +126,14 @@ public IBlockchainBridge CreateBlockchainBridge()
         public IManualBlockProductionTrigger ManualBlockProductionTrigger { get; set; } =
             new BuildBlocksWhenRequested();
 
-        public IIPResolver? IpResolver { get; set; }
-        public IJsonSerializer EthereumJsonSerializer { get; set; }
-        public IKeyStore? KeyStore { get; set; }
-        public IPasswordProvider? PasswordProvider { get; set; }
+        public IJsonSerializer EthereumJsonSerializer => BaseContainer.Resolve<IJsonSerializer>();
```

Comment:
Shouldn't those values be cached? Do we want to run `Resolve` on each call?

Reply:
Yea, I have the same thought. At the moment, I'm thinking, consumer should only be fetching from `INethermindApi` at start, so I'm just making things cleaner.